docs/book/src/cronjob-tutorial/testdata/project/api/v1/cronjob_webhook.go

```diff
@@ -0,0 +1,182 @@
+/*
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+// +kubebuilder:docs-gen:collapse=Apache License
+
+/*
+We'll start out with some imports.  You'll see below that we'll need a few more imports
+than those scaffolded for us.  We'll talk about each one when we use it.
+*/
+package v1
+
+import (
+	"github.com/robfig/cron"
+
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	apimachineryvalidation "k8s.io/apimachinery/pkg/api/validation"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	apimachineryutilvalidation "k8s.io/apimachinery/pkg/util/validation"
+	"k8s.io/apimachinery/pkg/util/validation/field"
+
+	logf "sigs.k8s.io/controller-runtime/pkg/runtime/log"
+	"sigs.k8s.io/controller-runtime/pkg/webhook"
+)
+
+/*
+Next, we'll setup logger for the webhooks.
```

Comment:
Use the log handle in the CronJob struct.  If we're not scaffolding that, that's a mistake.

Reply:
> Use the log handle in the CronJob struct.

Do you mean something like this?
```
type CronJob struct {
	metav1.TypeMeta   `json:",inline"`
	metav1.ObjectMeta `json:"metadata,omitempty"`

	Spec   CronJobSpec   `json:"spec,omitempty"`
	Status CronJobStatus `json:"status,omitempty"`

	log logr.Logger `json:"-"`
}
```
It's not clear to me how to wire the log handle when creating a new CronJob instance.
I'm not convinced putting a log handle in the struct is better. So please convince me :)

Reply:
An alternative is we always generate scaffold a package-level log handle in `groupversion_info.go` and just use that.

Reply:
oh, shoot, I misread.  nvm